packages/data/src/components/use-select/index.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * WordPress dependencies
+ */
+import { createQueue } from '@wordpress/priority-queue';
+import {
+	useLayoutEffect,
+	useRef,
+	useMemo,
+	useCallback,
+	useEffect,
+	useReducer,
+} from '@wordpress/element';
+import { isShallowEqualObjects } from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import useRegistry from '../registry-provider/use-registry';
+import useAsyncMode from '../async-mode-provider/use-async-mode';
+
+/**
+ * Favor useLayoutEffect to ensure the store subscription callback always has
```

Comment:
I really admire the emphasis in documentation in the changes of this pull request. 👍